src/hooks/LogHookContext.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HookData, RpcLogCategory, RpcLogLevel } from '@azure/functions-core';
+import * as coreTypes from '@azure/functions-core';
+import { AzureFunctionsRpcMessages as rpc } from '../../azure-functions-language-worker-protobuf/src/rpc';
+import { toCoreLog } from '../coreApi/converters/toCoreStatusResult';
+import { ReadOnlyError } from '../errors';
+import { nonNullProp } from '../utils/nonNull';
+import { worker } from '../WorkerContext';
+
+export interface InvocationLogContext {
+    hookData: HookData;
+    invocationContext: unknown;
+}
+
+export class LogHookContext implements coreTypes.LogHookContext {
+    level: RpcLogLevel;
+    message: string;
+    #hookData: HookData;
+    #category: RpcLogCategory;
+    #invocationContext: unknown;
+
+    constructor(log: rpc.IRpcLog, invocLogCtx: InvocationLogContext | undefined) {
+        this.#hookData = invocLogCtx?.hookData ?? {};
+        this.#invocationContext = invocLogCtx?.invocationContext;
+        const coreLog = toCoreLog(log);
+        this.level = nonNullProp(coreLog, 'level');
```

Comment:
Same nit here. I would rather all the private fields appear together at the beginning or at the end. This way seems inconsistent. I get why you did it this way, but I am okay with moving toCoreLog() to the top of it helps the ordering of what follows.

Reply:
I can update this as I agree it's a bit weird. I normally try to match the ordering of the properties above, which I obviously didn't do